src/Analyser/NodeScopeResolver.php

```diff
@@ -1410,10 +1409,16 @@ private function processStmtNode(
 				}
 			}
 
+			if ($alwaysIterates) {
+				$isAlwaysTerminating = count($finalScopeResult->getExitPointsByType(Break_::class)) === 0;
+			} else {
+				$isAlwaysTerminating = false; // $finalScopeResult->isAlwaysTerminating() && $isAlwaysIterable
```

Comment:
not a 100% sure what to do with this comment. it looks outdated too and should be most likely be `$isIterableAtLeastOnce->yes() && $finalScopeResult->isAlwaysTerminating()`. just worried that deleting it might lead to loosing _some_  old information

Reply:
actually doing this instead of setting it just to `false` leads to one failing tests:
> 1) PHPStan\Analyser\StatementResultTest::testIsAlwaysTerminating with data set #56 ('for ($i = 0; $i < 10; $i++) {...urn; }', false)
> Failed asserting that true is identical to false.

https://github.com/phpstan/phpstan-src/blob/eb6a95a9230d9c40e738c92edfef4f89e56678d2/tests/PHPStan/Analyser/StatementResultTest.php#L232-L235

but messing with this is out of scope of this MR anyway I'd say.

Reply:
I'd say this should be true so updating the test should be correct.